src/rules/at-root-no-redundant/index.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("at-root-no-redundant");
+
+function isWithinKeyframes(node) {
+  let parent = node.parent;
+  while (parent.type !== "root") {
```

Comment:
Nit: this condition is a little bit risky as it could throw an NPE if the node is "orphan". In practice it probably doesn't matter, but a safer check would be `while(parent)` or to be more explicit `while(parent != null)`